system/jlib/jsmartsock.cpp

```diff
@@ -242,13 +242,19 @@ CSmartSocketFactory::CSmartSocketFactory(IPropertyTree &service, bool _retry, un
     }
 }
 
-CSmartSocketFactory::CSmartSocketFactory(const char *_socklist, bool _retry, unsigned _retryInterval, unsigned _dnsInterval)
+CSmartSocketFactory::CSmartSocketFactory(const char *_socklist, IPropertyTree* _tlsConfig, bool _retry, unsigned _retryInterval, unsigned _dnsInterval)
 {
-    PROGLOG("CSmartSocketFactory::CSmartSocketFactory(%s)",_socklist?_socklist:"NULL");
+    PROGLOG("CSmartSocketFactory::CSmartSocketFactory(%s, tlsConfig(%s))",_socklist?_socklist:"NULL", _tlsConfig?"yes":"no");
     SmartSocketListParser slp(_socklist);
     if (slp.getSockets(sockArray) == 0)
         throw createSmartSocketException(0, "no endpoints defined");
 
+    if (_tlsConfig && _tlsConfig->queryName() && streq(_tlsConfig->queryName(), "ssl"))
```

Comment:
Why the check of the name against ssl?  It could do with a comment.

(For instance the value created with createIssuerTlsConfig doesn't have that name - but it isn't going through this code path...)

Reply:
The check does not add value on reconsideration, so I'll remove it and squash my commits. 

Initially it seemed reasonable to ensure we had a configuration for SSL/TLS but given the other constructor doesn't explicitly check and has a different format it makes little sense.